### ISSUE TITLE
failing tests erroring because of missing lldb

### ISSUE THREAD

**Issue by GitMensch** [rr-debugger/rr#3790]
I'm not sure what's going on - if those tests explicit check lldb, then those should be skipped if it isn't available as cmake only tests for an available GDB.

Additional information for the test runs: gdb is _not_ system installed but in an additional place that is in PATH (but that does work in all the other tests).

~~~
$> ctest --rerun-failed --output-on-failure

Test project /tmp/rr/build
      Start  924: alternate_thread_diversion
 3/14 Test  #924: alternate_thread_diversion .................***Failed  Error regular expression found in output. Regex=[FAILED]  0.95 sec
Test 'alternate_thread_diversion' PASSED (gdb)
FAILED: expecting "\(rr\)"
exception: End Of File (EOF). Braindead platform.
<pexpect.pty_spawn.spawn object at 0x7f1577f6d670>
command: /tmp/rr/build/bin/rr
args: [b'/tmp/rr/build/bin/rr', b'--suppress-environment-warnings', b'--fatal-errors', b'--check-cached-mmaps', b'--resource-path=/tmp/rr/build', b'replay', b'-d', b'lldb', b'-o', b'--no-use-colors']
buffer (last 100 chars): ''
before (last 100 chars): '[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.\r\n'
after: <class 'pexpect.exceptions.EOF'>
match: None
match_index: None
exitstatus: 1
flag_eof: True
pid: 863963
child_fd: 7
closed: False
timeout: 10000
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: <_io.TextIOWrapper name='lldb_rr.log' mode='w' encoding='UTF-8'>
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: re.compile('\\(rr\\)')
Test 'alternate_thread_diversion' FAILED: debug script failed (lldb)
--------------------------------------------------
lldb_rr.log:
[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.
--------------------------------------------------
debug.err:
cat: debug.err: No such file or directory
--------------------------------------------------
Test alternate_thread_diversion failed, leaving behind /tmp/rr-test-alternate_thread_diversion-1sksJvbWL and /tmp/rr/build/rr-test-alternate_thread_diversion-8JQGuUfZ6
To replay the failed test, run
  _RR_TRACE_DIR=/tmp/rr-test-alternate_thread_diversion-1sksJvbWL rr replay

      Start  925: alternate_thread_diversion-no-syscallbuf
 4/14 Test  #925: alternate_thread_diversion-no-syscallbuf ...***Failed  Error regular expression found in output. Regex=[FAILED]  0.92 sec
Test 'alternate_thread_diversion' PASSED (gdb)
FAILED: expecting "\(rr\)"
exception: End Of File (EOF). Braindead platform.
<pexpect.pty_spawn.spawn object at 0x7fa254192670>
command: /tmp/rr/build/bin/rr
args: [b'/tmp/rr/build/bin/rr', b'--suppress-environment-warnings', b'--fatal-errors', b'--check-cached-mmaps', b'--resource-path=/tmp/rr/build', b'replay', b'-d', b'lldb', b'-o', b'--no-use-colors']
buffer (last 100 chars): ''
before (last 100 chars): '[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.\r\n'
after: <class 'pexpect.exceptions.EOF'>
match: None
match_index: None
exitstatus: 1
flag_eof: True
pid: 864011
child_fd: 7
closed: False
timeout: 10000
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: <_io.TextIOWrapper name='lldb_rr.log' mode='w' encoding='UTF-8'>
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: re.compile('\\(rr\\)')
Test 'alternate_thread_diversion' FAILED: debug script failed (lldb)
--------------------------------------------------
lldb_rr.log:
[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.
--------------------------------------------------
debug.err:
cat: debug.err: No such file or directory
--------------------------------------------------
Test alternate_thread_diversion failed, leaving behind /tmp/rr-test-alternate_thread_diversion-F4l1PueIl and /tmp/rr/build/rr-test-alternate_thread_diversion-hzTTBwNYB
To replay the failed test, run
  _RR_TRACE_DIR=/tmp/rr-test-alternate_thread_diversion-F4l1PueIl rr replay

      Start  964: breakpoint
 5/14 Test  #964: breakpoint .................................***Failed  Error regular expression found in output. Regex=[FAILED]  0.90 sec
Test 'breakpoint' PASSED (gdb)
FAILED: expecting "\(rr\)"
exception: End Of File (EOF). Braindead platform.
<pexpect.pty_spawn.spawn object at 0x7f8bfc313940>
command: /tmp/rr/build/bin/rr
args: [b'/tmp/rr/build/bin/rr', b'--suppress-environment-warnings', b'--fatal-errors', b'--check-cached-mmaps', b'--resource-path=/tmp/rr/build', b'replay', b'-d', b'lldb', b'-o', b'--no-use-colors']
buffer (last 100 chars): ''
before (last 100 chars): '[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.\r\n'
after: <class 'pexpect.exceptions.EOF'>
match: None
match_index: None
exitstatus: 1
flag_eof: True
pid: 864055
child_fd: 7
closed: False
timeout: 10000
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: <_io.TextIOWrapper name='lldb_rr.log' mode='w' encoding='UTF-8'>
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: re.compile('\\(rr\\)')
Test 'breakpoint' FAILED: debug script failed (lldb)
--------------------------------------------------
lldb_rr.log:
[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.
--------------------------------------------------
debug.err:
cat: debug.err: No such file or directory
--------------------------------------------------
Test breakpoint failed, leaving behind /tmp/rr-test-breakpoint-ivV8WNt0w and /tmp/rr/build/rr-test-breakpoint-u5OU8L2f4
To replay the failed test, run
  _RR_TRACE_DIR=/tmp/rr-test-breakpoint-ivV8WNt0w rr replay

      Start  965: breakpoint-no-syscallbuf
 6/14 Test  #965: breakpoint-no-syscallbuf ...................***Failed  Error regular expression found in output. Regex=[FAILED]  0.90 sec
Test 'breakpoint' PASSED (gdb)
FAILED: expecting "\(rr\)"
exception: End Of File (EOF). Braindead platform.
<pexpect.pty_spawn.spawn object at 0x7f68c5dd3940>
command: /tmp/rr/build/bin/rr
args: [b'/tmp/rr/build/bin/rr', b'--suppress-environment-warnings', b'--fatal-errors', b'--check-cached-mmaps', b'--resource-path=/tmp/rr/build', b'replay', b'-d', b'lldb', b'-o', b'--no-use-colors']
buffer (last 100 chars): ''
before (last 100 chars): '[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.\r\n'
after: <class 'pexpect.exceptions.EOF'>
match: None
match_index: None
exitstatus: 1
flag_eof: True
pid: 864099
child_fd: 7
closed: False
timeout: 10000
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: <_io.TextIOWrapper name='lldb_rr.log' mode='w' encoding='UTF-8'>
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: re.compile('\\(rr\\)')
Test 'breakpoint' FAILED: debug script failed (lldb)
--------------------------------------------------
lldb_rr.log:
[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.
--------------------------------------------------
debug.err:
cat: debug.err: No such file or directory
--------------------------------------------------
Test breakpoint failed, leaving behind /tmp/rr-test-breakpoint-7HijHC68u and /tmp/rr/build/rr-test-breakpoint-roAz9Q39b
To replay the failed test, run
  _RR_TRACE_DIR=/tmp/rr-test-breakpoint-7HijHC68u rr replay

      Start 1294: watchpoint_unaligned
 7/14 Test #1294: watchpoint_unaligned .......................***Failed  Error regular expression found in output. Regex=[FAILED]  0.93 sec
Test 'watchpoint_unaligned' PASSED (gdb)
FAILED: expecting "\(rr\)"
exception: End Of File (EOF). Exception style platform.
<pexpect.pty_spawn.spawn object at 0x7f0e2d692040>
command: /tmp/rr/build/bin/rr
args: [b'/tmp/rr/build/bin/rr', b'--suppress-environment-warnings', b'--fatal-errors', b'--check-cached-mmaps', b'--resource-path=/tmp/rr/build', b'replay', b'-d', b'lldb', b'-o', b'--no-use-colors']
buffer (last 100 chars): ''
before (last 100 chars): '[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.\r\n'
after: <class 'pexpect.exceptions.EOF'>
match: None
match_index: None
exitstatus: None
flag_eof: True
pid: 864143
child_fd: 7
closed: False
timeout: 10000
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: <_io.TextIOWrapper name='lldb_rr.log' mode='w' encoding='UTF-8'>
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: re.compile('\\(rr\\)')
Test 'watchpoint_unaligned' FAILED: debug script failed (lldb)
--------------------------------------------------
lldb_rr.log:
[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.
--------------------------------------------------
debug.err:
cat: debug.err: No such file or directory
--------------------------------------------------
Test watchpoint_unaligned failed, leaving behind /tmp/rr-test-watchpoint_unaligned-kVRulhGal and /tmp/rr/build/rr-test-watchpoint_unaligned-Xjkzijqs5
To replay the failed test, run
  _RR_TRACE_DIR=/tmp/rr-test-watchpoint_unaligned-kVRulhGal rr replay

      Start 1295: watchpoint_unaligned-no-syscallbuf
 8/14 Test #1295: watchpoint_unaligned-no-syscallbuf .........***Failed  Error regular expression found in output. Regex=[FAILED]  0.94 sec
Test 'watchpoint_unaligned' PASSED (gdb)
FAILED: expecting "\(rr\)"
exception: End Of File (EOF). Braindead platform.
<pexpect.pty_spawn.spawn object at 0x7f8fc2ed3040>
command: /tmp/rr/build/bin/rr
args: [b'/tmp/rr/build/bin/rr', b'--suppress-environment-warnings', b'--fatal-errors', b'--check-cached-mmaps', b'--resource-path=/tmp/rr/build', b'replay', b'-d', b'lldb', b'-o', b'--no-use-colors']
buffer (last 100 chars): ''
before (last 100 chars): '[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.\r\n'
after: <class 'pexpect.exceptions.EOF'>
match: None
match_index: None
exitstatus: 1
flag_eof: True
pid: 864187
child_fd: 7
closed: False
timeout: 10000
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: <_io.TextIOWrapper name='lldb_rr.log' mode='w' encoding='UTF-8'>
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: re.compile('\\(rr\\)')
Test 'watchpoint_unaligned' FAILED: debug script failed (lldb)
--------------------------------------------------
lldb_rr.log:
[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.
--------------------------------------------------
debug.err:
cat: debug.err: No such file or directory
--------------------------------------------------
Test watchpoint_unaligned failed, leaving behind /tmp/rr-test-watchpoint_unaligned-L6rU5BbVq and /tmp/rr/build/rr-test-watchpoint_unaligned-LfONGpM89
To replay the failed test, run
  _RR_TRACE_DIR=/tmp/rr-test-watchpoint_unaligned-L6rU5BbVq rr replay

      Start 1296: when_threads
 9/14 Test #1296: when_threads ...............................***Failed  Error regular expression found in output. Regex=[FAILED]  1.91 sec
Test 'when_threads' PASSED (gdb)
FAILED: expecting "\(rr\)"
exception: End Of File (EOF). Braindead platform.
<pexpect.pty_spawn.spawn object at 0x7f87d39d7ca0>
command: /tmp/rr/build/bin/rr
args: [b'/tmp/rr/build/bin/rr', b'--suppress-environment-warnings', b'--fatal-errors', b'--check-cached-mmaps', b'--resource-path=/tmp/rr/build', b'replay', b'-d', b'lldb', b'-o', b'--no-use-colors']
buffer (last 100 chars): ''
before (last 100 chars): '[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.\r\n'
after: <class 'pexpect.exceptions.EOF'>
match: None
match_index: None
exitstatus: 1
flag_eof: True
pid: 864233
child_fd: 7
closed: False
timeout: 10000
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: <_io.TextIOWrapper name='lldb_rr.log' mode='w' encoding='UTF-8'>
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: re.compile('\\(rr\\)')
Test 'when_threads' FAILED: debug script failed (lldb)
--------------------------------------------------
lldb_rr.log:
[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.
--------------------------------------------------
debug.err:
cat: debug.err: No such file or directory
--------------------------------------------------
Test when_threads failed, leaving behind /tmp/rr-test-when_threads-i2qAf6tpI and /tmp/rr/build/rr-test-when_threads-wUzJMfW0G
To replay the failed test, run
  _RR_TRACE_DIR=/tmp/rr-test-when_threads-i2qAf6tpI rr replay

      Start 1297: when_threads-no-syscallbuf
10/14 Test #1297: when_threads-no-syscallbuf .................***Failed  Error regular expression found in output. Regex=[FAILED]  1.90 sec
Test 'when_threads' PASSED (gdb)
FAILED: expecting "\(rr\)"
exception: End Of File (EOF). Braindead platform.
<pexpect.pty_spawn.spawn object at 0x7f6419f57ca0>
command: /tmp/rr/build/bin/rr
args: [b'/tmp/rr/build/bin/rr', b'--suppress-environment-warnings', b'--fatal-errors', b'--check-cached-mmaps', b'--resource-path=/tmp/rr/build', b'replay', b'-d', b'lldb', b'-o', b'--no-use-colors']
buffer (last 100 chars): ''
before (last 100 chars): '[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.\r\n'
after: <class 'pexpect.exceptions.EOF'>
match: None
match_index: None
exitstatus: 1
flag_eof: True
pid: 864279
child_fd: 7
closed: False
timeout: 10000
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: <_io.TextIOWrapper name='lldb_rr.log' mode='w' encoding='UTF-8'>
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: re.compile('\\(rr\\)')
Test 'when_threads' FAILED: debug script failed (lldb)
--------------------------------------------------
lldb_rr.log:
[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.
--------------------------------------------------
debug.err:
cat: debug.err: No such file or directory
--------------------------------------------------
Test when_threads failed, leaving behind /tmp/rr-test-when_threads-hKOeAROc1 and /tmp/rr/build/rr-test-when_threads-hgVmzWqBw
To replay the failed test, run
  _RR_TRACE_DIR=/tmp/rr-test-when_threads-hKOeAROc1 rr replay

      Start 1330: breakpoint_print
11/14 Test #1330: breakpoint_print ...........................***Failed  Error regular expression found in output. Regex=[FAILED]  0.96 sec
Test 'breakpoint_print' PASSED (gdb)
FAILED: expecting "\(rr\)"
exception: End Of File (EOF). Braindead platform.
<pexpect.pty_spawn.spawn object at 0x7fc6c45d29a0>
command: /tmp/rr/build/bin/rr
args: [b'/tmp/rr/build/bin/rr', b'--suppress-environment-warnings', b'--fatal-errors', b'--check-cached-mmaps', b'--resource-path=/tmp/rr/build', b'replay', b'-d', b'lldb', b'-o', b'--no-use-colors']
buffer (last 100 chars): ''
before (last 100 chars): '[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.\r\n'
after: <class 'pexpect.exceptions.EOF'>
match: None
match_index: None
exitstatus: 1
flag_eof: True
pid: 864323
child_fd: 7
closed: False
timeout: 10000
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: <_io.TextIOWrapper name='lldb_rr.log' mode='w' encoding='UTF-8'>
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: re.compile('\\(rr\\)')
Test 'breakpoint_print' FAILED: debug script failed (lldb)
--------------------------------------------------
lldb_rr.log:
[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.
--------------------------------------------------
debug.err:
cat: debug.err: No such file or directory
--------------------------------------------------
Test breakpoint_print failed, leaving behind /tmp/rr-test-breakpoint_print-RTsMRUVtl and /tmp/rr/build/rr-test-breakpoint_print-uao0XWojW
To replay the failed test, run
  _RR_TRACE_DIR=/tmp/rr-test-breakpoint_print-RTsMRUVtl rr replay

      Start 1331: breakpoint_print-no-syscallbuf
12/14 Test #1331: breakpoint_print-no-syscallbuf .............***Failed  Error regular expression found in output. Regex=[FAILED]  0.89 sec
Test 'breakpoint_print' PASSED (gdb)
FAILED: expecting "\(rr\)"
exception: End Of File (EOF). Braindead platform.
<pexpect.pty_spawn.spawn object at 0x7fcb670ad9a0>
command: /tmp/rr/build/bin/rr
args: [b'/tmp/rr/build/bin/rr', b'--suppress-environment-warnings', b'--fatal-errors', b'--check-cached-mmaps', b'--resource-path=/tmp/rr/build', b'replay', b'-d', b'lldb', b'-o', b'--no-use-colors']
buffer (last 100 chars): ''
before (last 100 chars): '[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.\r\n'
after: <class 'pexpect.exceptions.EOF'>
match: None
match_index: None
exitstatus: 1
flag_eof: True
pid: 864367
child_fd: 7
closed: False
timeout: 10000
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: <_io.TextIOWrapper name='lldb_rr.log' mode='w' encoding='UTF-8'>
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: re.compile('\\(rr\\)')
Test 'breakpoint_print' FAILED: debug script failed (lldb)
--------------------------------------------------
lldb_rr.log:
[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.
--------------------------------------------------
debug.err:
cat: debug.err: No such file or directory
--------------------------------------------------
Test breakpoint_print failed, leaving behind /tmp/rr-test-breakpoint_print-o6d2pM0Co and /tmp/rr/build/rr-test-breakpoint_print-NHCEPy1Ut
To replay the failed test, run
  _RR_TRACE_DIR=/tmp/rr-test-breakpoint_print-o6d2pM0Co rr replay

      Start 1520: when
13/14 Test #1520: when .......................................***Failed  Error regular expression found in output. Regex=[FAILED]  0.93 sec
Test 'when' PASSED (gdb)
FAILED: expecting "\(rr\)"
exception: End Of File (EOF). Braindead platform.
<pexpect.pty_spawn.spawn object at 0x7f0f5e7d7f40>
command: /tmp/rr/build/bin/rr
args: [b'/tmp/rr/build/bin/rr', b'--suppress-environment-warnings', b'--fatal-errors', b'--check-cached-mmaps', b'--resource-path=/tmp/rr/build', b'replay', b'-d', b'lldb', b'-o', b'--no-use-colors']
buffer (last 100 chars): ''
before (last 100 chars): '[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.\r\n'
after: <class 'pexpect.exceptions.EOF'>
match: None
match_index: None
exitstatus: 1
flag_eof: True
pid: 864414
child_fd: 7
closed: False
timeout: 10000
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: <_io.TextIOWrapper name='lldb_rr.log' mode='w' encoding='UTF-8'>
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: re.compile('\\(rr\\)')
Test 'when' FAILED: debug script failed (lldb)
--------------------------------------------------
lldb_rr.log:
[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.
--------------------------------------------------
debug.err:
cat: debug.err: No such file or directory
--------------------------------------------------
Test when failed, leaving behind /tmp/rr-test-when-i5fFTqjIk and /tmp/rr/build/rr-test-when-E2dYtS7PP
To replay the failed test, run
  _RR_TRACE_DIR=/tmp/rr-test-when-i5fFTqjIk rr replay

      Start 1521: when-no-syscallbuf
14/14 Test #1521: when-no-syscallbuf .........................***Failed  Error regular expression found in output. Regex=[FAILED]  0.92 sec
Test 'when' PASSED (gdb)
FAILED: expecting "\(rr\)"
exception: End Of File (EOF). Braindead platform.
<pexpect.pty_spawn.spawn object at 0x7f5fbff17f40>
command: /tmp/rr/build/bin/rr
args: [b'/tmp/rr/build/bin/rr', b'--suppress-environment-warnings', b'--fatal-errors', b'--check-cached-mmaps', b'--resource-path=/tmp/rr/build', b'replay', b'-d', b'lldb', b'-o', b'--no-use-colors']
buffer (last 100 chars): ''
before (last 100 chars): '[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.\r\n'
after: <class 'pexpect.exceptions.EOF'>
match: None
match_index: None
exitstatus: 1
flag_eof: True
pid: 864461
child_fd: 7
closed: False
timeout: 10000
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: <_io.TextIOWrapper name='lldb_rr.log' mode='w' encoding='UTF-8'>
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: re.compile('\\(rr\\)')
Test 'when' FAILED: debug script failed (lldb)
--------------------------------------------------
lldb_rr.log:
[FATAL src/launch_debugger.cc:356:launch_debugger()] Failed to exec lldb.
--------------------------------------------------
debug.err:
cat: debug.err: No such file or directory
--------------------------------------------------
Test when failed, leaving behind /tmp/rr-test-when-RoZqBdmAv and /tmp/rr/build/rr-test-when-nVZBbJfO1
To replay the failed test, run
  _RR_TRACE_DIR=/tmp/rr-test-when-RoZqBdmAv rr replay


0% tests passed, 14 tests failed out of 14

Total Test time (real) =  13.66 sec

The following tests FAILED:
        924 - alternate_thread_diversion (Failed)
        925 - alternate_thread_diversion-no-syscallbuf (Failed)
        964 - breakpoint (Failed)
        965 - breakpoint-no-syscallbuf (Failed)
        1294 - watchpoint_unaligned (Failed)
        1295 - watchpoint_unaligned-no-syscallbuf (Failed)
        1296 - when_threads (Failed)
        1297 - when_threads-no-syscallbuf (Failed)
        1330 - breakpoint_print (Failed)
        1331 - breakpoint_print-no-syscallbuf (Failed)
        1520 - when (Failed)
        1521 - when-no-syscallbuf (Failed)
Errors while running CTest
~~~

**Comment by rocallahan** [comment]
I think we'll just require lldb to be installed as well as gdb if you're going to run tests.

**Comment by GitMensch** [comment]
This definitely resolves the issue but - whoa, LLDB has clang dependency meaning 100MB+ download on an already setup RHEL environment with around 400 MB install size.

For comparison: the other build dependencies got in with 1 MB download / 3 MB install size plus additional 10 MB / 38 MB for cmake.

I guess you would be fine with a PR that skips those tests instead if LLDB is not available?
Should this be cmake configured or skipping "dynamically? Is there something one could use as outline to do so?

**Comment by rocallahan** [comment]
I really don't think it's too much to ask for a 100MB download to run rr's tests.

**Comment by GitMensch** [comment]
Should we reopen this? Using current rr master and the same machine, now WITH lldb, fails in exactly the same tests...

~~~

        935 - alternate_thread_diversion (Failed)
        936 - alternate_thread_diversion-no-syscallbuf (Failed)
        975 - breakpoint (Failed)
        976 - breakpoint-no-syscallbuf (Failed)
        1313 - x86/watchpoint_error (Failed)
        1314 - x86/watchpoint_error-no-syscallbuf (Failed)
        1321 - watchpoint_unaligned (Failed)
        1322 - watchpoint_unaligned-no-syscallbuf (Failed)
        1323 - when_threads (Failed)
        1324 - when_threads-no-syscallbuf (Failed)
        1357 - breakpoint_print (Failed)
        1358 - breakpoint_print-no-syscallbuf (Failed)
        1551 - when (Failed)
        1552 - when-no-syscallbuf (Failed)
~~~

That's the output from the first test:

`lldb_rr.log`:
~~~
 cat /tmp/rr-test-alternate_thread_diversion-QXYOuvsuO/lldb_rr.log
(lldb) target create "/tmp/rr-test-alternate_thread_diversion-QXYOuvsuO/alternate_thread_diversion-QXYOuvsuO-0/mmap_hardlink_4_alternate_thread_diversion-QXYOuvsuO"
Current executable set to '/tmp/rr-test-alternate_thread_diversion-QXYOuvsuO/alternate_thread_diversion-QXYOuvsuO-0/mmap_hardlink_4_alternate_thread_diversion-QXYOuvsuO' (x86_64).
(lldb) script exec(open('/proc/10702/fd/5').read())
(rr) gdb-remote [127.0.0.1]:10703
command source -s 0 /home/build/rr/src/test/test_setup.lldb
breakpoint set --name break_here
command source -s 0 /home/build/rr/src/test/test_setup.lldb
breakpoint set --name break_here
Process 10679 stopped
(rr) command source -s 0 /home/build/rr/src/test/test_setup.lldb
Executing commands in '/home/build/rr/src/test/test_setup.lldb'.
(rr) process handle -p true -s true SIGSEGV
NAME         PASS   STOP   NOTIFY
===========  =====  =====  ======
SIGSEGV      true   true   true
(rr) process handle -p false -s false SIGKILL
NAME         PASS   STOP   NOTIFY
===========  =====  =====  ======
SIGKILL      false  false  true
(rr) breakpoint set --name break_here
Breakpoint 1: where = mmap_hardlink_4_alternate_thread_diversion-QXYOuvsuO`break_here + 4 at alternate_thread_diversion.c:7:71, address = 0x000055b7608f71ac
(rr) continue
continue
Process 10679 resuming
Process 10679 stopped
* thread #1, stop reason = signal SIGTRAP
    frame #0: 0xffffffffffffffff
~~~
and `test-monitor.output`: 
~~~
$> cat /tmp/rr-test-alternate_thread_diversion-QXYOuvsuO/test-monitor.output
timeout 120 exceeded
====== /proc/10701/status
Name:   python3
Umask:  0022
State:  S (sleeping)
Tgid:   10701
Ngid:   0
Pid:    10701
PPid:   10700
TracerPid:      0
Uid:    1000    1000    1000    1000
Gid:    1000    1000    1000    1000
FDSize: 64
Groups: 4 24 27 30 46 1000
NStgid: 10701
NSpid:  10701
NSpgid: 10701
NSsid:  10
VmPeak:    14336 kB
VmSize:    14336 kB
VmLck:         0 kB
VmPin:         0 kB
VmHWM:     11104 kB
VmRSS:     11104 kB
RssAnon:            4732 kB
RssFile:            6372 kB
RssShmem:              0 kB
VmData:     5000 kB
VmStk:       132 kB
VmExe:      2596 kB
VmLib:      2568 kB
VmPTE:        60 kB
VmSwap:        0 kB
HugetlbPages:          0 kB
CoreDumping:    0
THP_enabled:    1
Threads:        1
SigQ:   1/31181
SigPnd: 0000000000000000
ShdPnd: 0000000000000000
SigBlk: 0000000000000000
SigIgn: 0000000001001000
SigCgt: 0000000180000002
CapInh: 0000000000000000
CapPrm: 0000000000000000
CapEff: 0000000000000000
CapBnd: 000001ffffffffff
CapAmb: 0000000000000000
NoNewPrivs:     0
Seccomp:        0
Seccomp_filters:        0
Speculation_Store_Bypass:       thread vulnerable
SpeculationIndirectBranch:      conditional enabled
Cpus_allowed:   ff
Cpus_allowed_list:      0-7
Mems_allowed:   1
Mems_allowed_list:      0
voluntary_ctxt_switches:        46
nonvoluntary_ctxt_switches:     0
====== /proc/10701/stack
====== /proc/10702/status
Name:   lldb
Umask:  0022
State:  S (sleeping)
Tgid:   10702
Ngid:   0
Pid:    10702
PPid:   10701
TracerPid:      0
Uid:    1000    1000    1000    1000
Gid:    1000    1000    1000    1000
FDSize: 64
Groups: 4 24 27 30 46 1000
NStgid: 10702
NSpid:  10702
NSpgid: 10702
NSsid:  10702
VmPeak:   945528 kB
VmSize:   912744 kB
VmLck:         0 kB
VmPin:         0 kB
VmHWM:     89120 kB
VmRSS:     89108 kB
RssAnon:           37960 kB
RssFile:           51148 kB
RssShmem:              0 kB
VmData:    81664 kB
VmStk:       132 kB
VmExe:        48 kB
VmLib:    160688 kB
VmPTE:       424 kB
VmSwap:        0 kB
HugetlbPages:          0 kB
CoreDumping:    0
THP_enabled:    1
Threads:        4
SigQ:   1/31181
SigPnd: 0000000000000000
ShdPnd: 0000000000000000
SigBlk: 0000000000000000
SigIgn: 0000000000001000
SigCgt: 00000001c98a4eff
CapInh: 0000000000000000
CapPrm: 0000000000000000
CapEff: 0000000000000000
CapBnd: 000001ffffffffff
CapAmb: 0000000000000000
NoNewPrivs:     0
Seccomp:        0
Seccomp_filters:        0
Speculation_Store_Bypass:       thread vulnerable
SpeculationIndirectBranch:      conditional enabled
Cpus_allowed:   ff
Cpus_allowed_list:      0-7
Mems_allowed:   1
Mems_allowed_list:      0
voluntary_ctxt_switches:        257
nonvoluntary_ctxt_switches:     0
====== /proc/10702/stack
====== /proc/10717/status
Name:   b-remote.async>
Umask:  0022
State:  S (sleeping)
Tgid:   10702
Ngid:   0
Pid:    10717
PPid:   10701
TracerPid:      0
Uid:    1000    1000    1000    1000
Gid:    1000    1000    1000    1000
FDSize: 64
Groups: 4 24 27 30 46 1000
NStgid: 10702
NSpid:  10717
NSpgid: 10702
NSsid:  10702
VmPeak:   945528 kB
VmSize:   912744 kB
VmLck:         0 kB
VmPin:         0 kB
VmHWM:     89120 kB
VmRSS:     89108 kB
RssAnon:           37960 kB
RssFile:           51148 kB
RssShmem:              0 kB
VmData:    81664 kB
VmStk:       132 kB
VmExe:        48 kB
VmLib:    160688 kB
VmPTE:       424 kB
VmSwap:        0 kB
HugetlbPages:          0 kB
CoreDumping:    0
THP_enabled:    1
Threads:        4
SigQ:   1/31181
SigPnd: 0000000000000000
ShdPnd: 0000000000000000
SigBlk: 0000000000000000
SigIgn: 0000000000001000
SigCgt: 00000001c98a4eff
CapInh: 0000000000000000
CapPrm: 0000000000000000
CapEff: 0000000000000000
CapBnd: 000001ffffffffff
CapAmb: 0000000000000000
NoNewPrivs:     0
Seccomp:        0
Seccomp_filters:        0
Speculation_Store_Bypass:       thread vulnerable
SpeculationIndirectBranch:      conditional enabled
Cpus_allowed:   ff
Cpus_allowed_list:      0-7
Mems_allowed:   1
Mems_allowed_list:      0
voluntary_ctxt_switches:        3
nonvoluntary_ctxt_switches:     0
====== /proc/10717/stack
====== /proc/10726/status
Name:   intern-state
Umask:  0022
State:  S (sleeping)
Tgid:   10702
Ngid:   0
Pid:    10726
PPid:   10701
TracerPid:      0
Uid:    1000    1000    1000    1000
Gid:    1000    1000    1000    1000
FDSize: 64
Groups: 4 24 27 30 46 1000
NStgid: 10702
NSpid:  10726
NSpgid: 10702
NSsid:  10702
VmPeak:   945528 kB
VmSize:   912744 kB
VmLck:         0 kB
VmPin:         0 kB
VmHWM:     89120 kB
VmRSS:     89108 kB
RssAnon:           37960 kB
RssFile:           51148 kB
RssShmem:              0 kB
VmData:    81664 kB
VmStk:       132 kB
VmExe:        48 kB
VmLib:    160688 kB
VmPTE:       424 kB
VmSwap:        0 kB
HugetlbPages:          0 kB
CoreDumping:    0
THP_enabled:    1
Threads:        4
SigQ:   1/31181
SigPnd: 0000000000000000
ShdPnd: 0000000000000000
SigBlk: 0000000000000000
SigIgn: 0000000000001000
SigCgt: 00000001c98a4eff
CapInh: 0000000000000000
CapPrm: 0000000000000000
CapEff: 0000000000000000
CapBnd: 000001ffffffffff
CapAmb: 0000000000000000
NoNewPrivs:     0
Seccomp:        0
Seccomp_filters:        0
Speculation_Store_Bypass:       thread vulnerable
SpeculationIndirectBranch:      conditional enabled
Cpus_allowed:   ff
Cpus_allowed_list:      0-7
Mems_allowed:   1
Mems_allowed_list:      0
voluntary_ctxt_switches:        3
nonvoluntary_ctxt_switches:     0
====== /proc/10726/stack
====== /proc/10727/status
Name:   dbg.evt-handler
Umask:  0022
State:  S (sleeping)
Tgid:   10702
Ngid:   0
Pid:    10727
PPid:   10701
TracerPid:      0
Uid:    1000    1000    1000    1000
Gid:    1000    1000    1000    1000
FDSize: 64
Groups: 4 24 27 30 46 1000
NStgid: 10702
NSpid:  10727
NSpgid: 10702
NSsid:  10702
VmPeak:   945528 kB
VmSize:   912744 kB
VmLck:         0 kB
VmPin:         0 kB
VmHWM:     89120 kB
VmRSS:     89108 kB
RssAnon:           37960 kB
RssFile:           51148 kB
RssShmem:              0 kB
VmData:    81664 kB
VmStk:       132 kB
VmExe:        48 kB
VmLib:    160688 kB
VmPTE:       424 kB
VmSwap:        0 kB
HugetlbPages:          0 kB
CoreDumping:    0
THP_enabled:    1
Threads:        4
SigQ:   1/31181
SigPnd: 0000000000000000
ShdPnd: 0000000000000000
SigBlk: 0000000000000000
SigIgn: 0000000000001000
SigCgt: 00000001c98a4eff
CapInh: 0000000000000000
CapPrm: 0000000000000000
CapEff: 0000000000000000
CapBnd: 000001ffffffffff
CapAmb: 0000000000000000
NoNewPrivs:     0
Seccomp:        0
Seccomp_filters:        0
Speculation_Store_Bypass:       thread vulnerable
SpeculationIndirectBranch:      conditional enabled
Cpus_allowed:   ff
Cpus_allowed_list:      0-7
Mems_allowed:   1
Mems_allowed_list:      0
voluntary_ctxt_switches:        6
nonvoluntary_ctxt_switches:     0
====== /proc/10727/stack
====== /proc/10703/status
Name:   rr
Umask:  0022
State:  S (sleeping)
Tgid:   10703
Ngid:   0
Pid:    10703
PPid:   10702
TracerPid:      0
Uid:    1000    1000    1000    1000
Gid:    1000    1000    1000    1000
FDSize: 64
Groups: 4 24 27 30 46 1000
NStgid: 10703
NSpid:  10703
NSpgid: 10702
NSsid:  10702
VmPeak:    17380 kB
VmSize:    17332 kB
VmLck:         0 kB
VmPin:         0 kB
VmHWM:      9580 kB
VmRSS:      9580 kB
RssAnon:            1236 kB
RssFile:            8344 kB
RssShmem:              0 kB
VmData:     1184 kB
VmStk:       132 kB
VmExe:      3236 kB
VmLib:      4664 kB
VmPTE:        76 kB
VmSwap:        0 kB
HugetlbPages:          0 kB
CoreDumping:    0
THP_enabled:    1
Threads:        1
SigQ:   1/31181
SigPnd: 0000000000000000
ShdPnd: 0000000000000000
SigBlk: 0000000000000000
SigIgn: 0000000001001000
SigCgt: 0000000180002002
CapInh: 0000000000000000
CapPrm: 0000000000000000
CapEff: 0000000000000000
CapBnd: 000001ffffffffff
CapAmb: 0000000000000000
NoNewPrivs:     0
Seccomp:        0
Seccomp_filters:        0
Speculation_Store_Bypass:       thread vulnerable
SpeculationIndirectBranch:      conditional enabled
Cpus_allowed:   40
Cpus_allowed_list:      6
Mems_allowed:   1
Mems_allowed_list:      0
voluntary_ctxt_switches:        323
nonvoluntary_ctxt_switches:     0
====== /proc/10703/stack
====== /proc/10704/status
Name:   rr:alternate_th
Umask:  0022
State:  t (tracing stop)
Tgid:   10704
Ngid:   0
Pid:    10704
PPid:   10703
TracerPid:      10703
Uid:    1000    1000    1000    1000
Gid:    1000    1000    1000    1000
FDSize: 1024
Groups: 4 24 27 30 46 1000
NStgid: 10704
NSpid:  10704
NSpgid: 10704
NSsid:  10704
VmPeak:     2408 kB
VmSize:     2408 kB
VmLck:         0 kB
VmPin:         0 kB
VmHWM:       300 kB
VmRSS:       300 kB
RssAnon:             168 kB
RssFile:             132 kB
RssShmem:              0 kB
VmData:     2204 kB
VmStk:         0 kB
VmExe:         4 kB
VmLib:       152 kB
VmPTE:        48 kB
VmSwap:        0 kB
HugetlbPages:          0 kB
CoreDumping:    0
THP_enabled:    1
Threads:        1
SigQ:   1/31181
SigPnd: 0000000000000000
ShdPnd: 0000000000000000
SigBlk: 0000000000000000
SigIgn: 0000000000010000
SigCgt: 0000000000000000
CapInh: 0000000000000000
CapPrm: 0000000000000000
CapEff: 0000000000000000
CapBnd: 000001ffffffffff
CapAmb: 0000000000000000
NoNewPrivs:     1
Seccomp:        0
Seccomp_filters:        0
Speculation_Store_Bypass:       thread vulnerable
SpeculationIndirectBranch:      conditional enabled
Cpus_allowed:   40
Cpus_allowed_list:      6
Mems_allowed:   1
Mems_allowed_list:      0
voluntary_ctxt_switches:        76
nonvoluntary_ctxt_switches:     0
====== /proc/10704/stack
====== /proc/10705/status
Name:   rr:alternate_th
Umask:  0022
State:  t (tracing stop)
Tgid:   10705
Ngid:   0
Pid:    10705
PPid:   10704
TracerPid:      10703
Uid:    1000    1000    1000    1000
Gid:    1000    1000    1000    1000
FDSize: 1024
Groups: 4 24 27 30 46 1000
NStgid: 10705
NSpid:  10705
NSpgid: 10704
NSsid:  10704
VmPeak:     2408 kB
VmSize:     2408 kB
VmLck:         0 kB
VmPin:         0 kB
VmHWM:       156 kB
VmRSS:       156 kB
RssAnon:             152 kB
RssFile:               4 kB
RssShmem:              0 kB
VmData:     2204 kB
VmStk:         0 kB
VmExe:         4 kB
VmLib:       152 kB
VmPTE:        44 kB
VmSwap:        0 kB
HugetlbPages:          0 kB
CoreDumping:    0
THP_enabled:    1
Threads:        1
SigQ:   1/31181
SigPnd: 0000000000000000
ShdPnd: 0000000000000000
SigBlk: 0000000000000000
SigIgn: 0000000000010000
SigCgt: 0000000000000000
CapInh: 0000000000000000
CapPrm: 0000000000000000
CapEff: 0000000000000000
CapBnd: 000001ffffffffff
CapAmb: 0000000000000000
NoNewPrivs:     1
Seccomp:        0
Seccomp_filters:        0
Speculation_Store_Bypass:       thread vulnerable
SpeculationIndirectBranch:      conditional enabled
Cpus_allowed:   40
Cpus_allowed_list:      6
Mems_allowed:   1
Mems_allowed_list:      0
voluntary_ctxt_switches:        4
nonvoluntary_ctxt_switches:     0
====== /proc/10705/stack
====== /proc/10716/status
Name:   rr:alternate_th
Umask:  0022
State:  t (tracing stop)
Tgid:   10716
Ngid:   0
Pid:    10716
PPid:   10704
TracerPid:      10703
Uid:    1000    1000    1000    1000
Gid:    1000    1000    1000    1000
FDSize: 1024
Groups: 4 24 27 30 46 1000
NStgid: 10716
NSpid:  10716
NSpgid: 10704
NSsid:  10704
VmPeak:     2408 kB
VmSize:     2408 kB
VmLck:         0 kB
VmPin:         0 kB
VmHWM:       156 kB
VmRSS:       156 kB
RssAnon:             152 kB
RssFile:               4 kB
RssShmem:              0 kB
VmData:     2204 kB
VmStk:         0 kB
VmExe:         4 kB
VmLib:       152 kB
VmPTE:        44 kB
VmSwap:        0 kB
HugetlbPages:          0 kB
CoreDumping:    0
THP_enabled:    1
Threads:        1
SigQ:   1/31181
SigPnd: 0000000000000000
ShdPnd: 0000000000000000
SigBlk: 0000000000000000
SigIgn: 0000000000010000
SigCgt: 0000000000000000
CapInh: 0000000000000000
CapPrm: 0000000000000000
CapEff: 0000000000000000
CapBnd: 000001ffffffffff
CapAmb: 0000000000000000
NoNewPrivs:     1
Seccomp:        0
Seccomp_filters:        0
Speculation_Store_Bypass:       thread vulnerable
SpeculationIndirectBranch:      conditional enabled
Cpus_allowed:   40
Cpus_allowed_list:      6
Mems_allowed:   1
Mems_allowed_list:      0
voluntary_ctxt_switches:        4
nonvoluntary_ctxt_switches:     0
====== /proc/10716/stack
====== gdb -p 10703 -ex 'set confirm off' -ex 'set height 0' -ex 'thread apply all bt' -ex q </dev/null 2>&1
GNU gdb (Debian 10.1-1.7) 10.1.90.20210103-git
Copyright (C) 2021 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word".
Attaching to process 10703
Reading symbols from /home/build/rr/build/bin/rr...
Reading symbols from /lib/x86_64-linux-gnu/libdl.so.2...
Reading symbols from /usr/lib/debug/.build-id/46/b3bf3f9b9eb092a5c0cf5575e89092f768054c.debug...
Reading symbols from /lib/x86_64-linux-gnu/libz.so.1...
(No debugging symbols found in /lib/x86_64-linux-gnu/libz.so.1)
Reading symbols from /lib/x86_64-linux-gnu/libzstd.so.1...
(No debugging symbols found in /lib/x86_64-linux-gnu/libzstd.so.1)
Reading symbols from /lib/x86_64-linux-gnu/libcapnp-0.7.0.so...
(No debugging symbols found in /lib/x86_64-linux-gnu/libcapnp-0.7.0.so)
Reading symbols from /lib/x86_64-linux-gnu/libkj-0.7.0.so...
(No debugging symbols found in /lib/x86_64-linux-gnu/libkj-0.7.0.so)
Reading symbols from /lib/x86_64-linux-gnu/libpthread.so.0...
Reading symbols from /usr/lib/debug/.build-id/25/5e355c207aba91a59ae1f808e3b4da443abf0c.debug...
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Reading symbols from /lib/x86_64-linux-gnu/libstdc++.so.6...
(No debugging symbols found in /lib/x86_64-linux-gnu/libstdc++.so.6)
Reading symbols from /lib/x86_64-linux-gnu/libm.so.6...
Reading symbols from /usr/lib/debug/.build-id/1d/6ff6c4c69f3572486bc27b8290ee932b0b9f39.debug...
Reading symbols from /lib/x86_64-linux-gnu/libgcc_s.so.1...
(No debugging symbols found in /lib/x86_64-linux-gnu/libgcc_s.so.1)
Reading symbols from /lib/x86_64-linux-gnu/libc.so.6...
Reading symbols from /usr/lib/debug/.build-id/f3/654f4d10c3f54ac5dabcb8fc67a2b8d1409dc4.debug...
Reading symbols from /lib64/ld-linux-x86-64.so.2...
Reading symbols from /usr/lib/debug/.build-id/1b/3277a419c3fa42b199e5a170ea215b32689793.debug...
Reading symbols from /lib/x86_64-linux-gnu/libthread_db.so.1...
Reading symbols from /usr/lib/debug/.build-id/58/a762c36aad815f366a353d32b0752a77be4d61.debug...
0x00007f81c1fe09d3 in __GI___poll (fds=0x7ffc3d82f00c, nfds=1, timeout=-1)
    at ../sysdeps/unix/sysv/linux/poll.c:29
29      ../sysdeps/unix/sysv/linux/poll.c: No such file or directory.

Thread 1 (Thread 0x7f81c1eec740 (LWP 10703) "rr"):
#0  0x00007f81c1fe09d3 in __GI___poll (fds=0x7ffc3d82f00c, nfds=1, timeout=-1) at ../sysdeps/unix/sysv/linux/poll.c:29
#1  0x000055d292ee85a7 in rr::poll_socket (sock_fd=..., events=1, timeoutMs=-1) at /home/build/rr/src/GdbServerConnection.cc:137
#2  0x000055d292ee864b in rr::poll_incoming (sock_fd=..., timeoutMs=-1) at /home/build/rr/src/GdbServerConnection.cc:145
#3  0x000055d292ee86a3 in rr::GdbServerConnection::read_data_once (this=0x55d2cffbf540) at /home/build/rr/src/GdbServerConnection.cc:159
#4  0x000055d292ee946d in rr::GdbServerConnection::read_packet (this=0x55d2cffbf540) at /home/build/rr/src/GdbServerConnection.cc:360
#5  0x000055d292ef026e in rr::GdbServerConnection::get_request (this=0x55d2cffbf540) at /home/build/rr/src/GdbServerConnection.cc:1614
#6  0x000055d292f0990e in rr::GdbServer::process_debugger_requests (this=0x7ffc3d8317c0, state=rr::GdbServer::REPORT_NORMAL) at /home/build/rr/src/GdbServer.cc:1239
#7  0x000055d292f0ab20 in rr::GdbServer::debug_one_step (this=0x7ffc3d8317c0, last_resume_request=...) at /home/build/rr/src/GdbServer.cc:1432
#8  0x000055d292f0cee9 in rr::GdbServer::serve_replay (session=std::shared_ptr<rr::ReplaySession> (empty) = {...}, target=..., stop_replaying_to_target=0x55d2933448b0 <rr::stop_replaying_to_target>, flags=...) at /home/build/rr/src/GdbServer.cc:1864
#9  0x000055d293043490 in rr::replay (trace_dir="", flags=...) at /home/build/rr/src/ReplayCommand.cc:553
#10 0x000055d293043e59 in rr::ReplayCommand::run (this=0x55d293344890 <rr::ReplayCommand::singleton>, args=std::vector of length 0, capacity 16) at /home/build/rr/src/ReplayCommand.cc:682
#11 0x000055d292f5eaeb in main (argc=10, argv=0x7ffc3d8334a8) at /home/build/rr/src/main.cc:278
Detaching from program: /home/build/rr/build/bin/rr, process 10703
[Inferior 1 (process 10703) detached]
====== gdb -p 10703 -ex 'set confirm off' -ex 'set height 0' -ex 'b rr::force_close_record_session' -ex 'p rr::force_close_record_session()' -ex detach -ex q </dev/null 2>&1
GNU gdb (Debian 10.1-1.7) 10.1.90.20210103-git
Copyright (C) 2021 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word".
Attaching to process 10703
Reading symbols from /home/build/rr/build/bin/rr...
Reading symbols from /lib/x86_64-linux-gnu/libdl.so.2...
Reading symbols from /usr/lib/debug/.build-id/46/b3bf3f9b9eb092a5c0cf5575e89092f768054c.debug...
Reading symbols from /lib/x86_64-linux-gnu/libz.so.1...
(No debugging symbols found in /lib/x86_64-linux-gnu/libz.so.1)
Reading symbols from /lib/x86_64-linux-gnu/libzstd.so.1...
(No debugging symbols found in /lib/x86_64-linux-gnu/libzstd.so.1)
Reading symbols from /lib/x86_64-linux-gnu/libcapnp-0.7.0.so...
(No debugging symbols found in /lib/x86_64-linux-gnu/libcapnp-0.7.0.so)
Reading symbols from /lib/x86_64-linux-gnu/libkj-0.7.0.so...
(No debugging symbols found in /lib/x86_64-linux-gnu/libkj-0.7.0.so)
Reading symbols from /lib/x86_64-linux-gnu/libpthread.so.0...
Reading symbols from /usr/lib/debug/.build-id/25/5e355c207aba91a59ae1f808e3b4da443abf0c.debug...
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Reading symbols from /lib/x86_64-linux-gnu/libstdc++.so.6...
(No debugging symbols found in /lib/x86_64-linux-gnu/libstdc++.so.6)
Reading symbols from /lib/x86_64-linux-gnu/libm.so.6...
Reading symbols from /usr/lib/debug/.build-id/1d/6ff6c4c69f3572486bc27b8290ee932b0b9f39.debug...
Reading symbols from /lib/x86_64-linux-gnu/libgcc_s.so.1...
(No debugging symbols found in /lib/x86_64-linux-gnu/libgcc_s.so.1)
Reading symbols from /lib/x86_64-linux-gnu/libc.so.6...
Reading symbols from /usr/lib/debug/.build-id/f3/654f4d10c3f54ac5dabcb8fc67a2b8d1409dc4.debug...
Reading symbols from /lib64/ld-linux-x86-64.so.2...
Reading symbols from /usr/lib/debug/.build-id/1b/3277a419c3fa42b199e5a170ea215b32689793.debug...
Reading symbols from /lib/x86_64-linux-gnu/libthread_db.so.1...
Reading symbols from /usr/lib/debug/.build-id/58/a762c36aad815f366a353d32b0752a77be4d61.debug...
0x00007f81c1fe09d3 in __GI___poll (fds=0x7ffc3d82f00c, nfds=1, timeout=-1)
    at ../sysdeps/unix/sysv/linux/poll.c:29
29      ../sysdeps/unix/sysv/linux/poll.c: No such file or directory.
Breakpoint 1 at 0x55d292f9bc4d: file /home/build/rr/src/RecordCommand.cc, line 628.

Breakpoint 1, rr::force_close_record_session ()
    at /home/build/rr/src/RecordCommand.cc:628
628       if (static_session) {
The program being debugged stopped while in a function called from GDB.
Evaluation of the expression containing the function
(rr::force_close_record_session()) will be abandoned.
When the function is done executing, GDB will silently stop.
Detaching from program: /home/build/rr/build/bin/rr, process 10703
[Inferior 1 (process 10703) detached]
~~~

**Comment by rocallahan** [comment]
We shouldn't reopen this issue, because your problem is clearly a different issue related to LLDB not working properly on your system, at least when it's talking to rr.
```
* thread #1, stop reason = signal SIGTRAP
```
The stop reason should be `breakpoint`. You can open a new issue for that, but I'm not going to work on it myself since I'm swamped.

**Comment by GitMensch** [comment]
I'd still vote for a way to make lldb optional - which reduces the dependencies and in this case ... failed tests.
... I really only care about GDB in my environments (and where lldb is a nice option to replace it rr doesn't work).

Sadly "lldb" is not in the tests' name, otherwise I could disable those with `ctest -E lldb` - is there another option to skip those?

**Comment by rocallahan** [comment]
rr's tests are mainly intended for rr developers. For us, it's important to test lldb so we don't regress with it.

I think you're describing a different use-case for rr's tests --- determine whether rr will work for you. TBH I don't think running rr's tests is actually a great fit for that, because a lot of them test obscure stuff that won't affect most developers.
